lib/streamlit/state/widgets.py

```diff
@@ -268,16 +271,14 @@ def _get_widget_id(
 ) -> str:
     """Generate a widget id for the given widget.
 
-    If user_key is defined, the widget_id returned is simply user_key.
-    Otherwise, we return a hash of the widget element type and the
-    string-serialized widget proto.
+    The widget id includes the user_key, which may be `None`, so widgets with
```

Comment:
I don't fully understand this comment. If we have two of the same widgets with the same arguments and the user doesn't provide a user_key will the widget ids be the same?

Reply:
Yes (and this will result in an exception when the second is created)

Reply:
OK, that makes sense!